File: make_graph.m
load("bi_result.csv")
load("new_result.csv")

figure;
set(0,'defaultAxesFontSize',25);
set(0,'defaultAxesFontName','Times New Roman');
set(0,'defaultTextFontSize',25);
set(0,'defaultTextFontName','Times New Roman');


subplot(2,1,1);

plot(bi_result(:,1),bi_result(:,3),'-b','Linewidth',2);
hold on;
plot(new_result(:,1),new_result(:,3),'-r','Linewidth',2);

xlabel("繰り返し回数: {\it n}");

ylabel("推定解: {\it x_n}");

legend('２分法','ニュートン法')

grid on;

subplot(2,1,2);

semilogy(bi_result(:,1),bi_result(:,2),'-b','Linewidth',2);
hold on;
semilogy(new_result(:,1),new_result(:,2),'-r','Linewidth',2);

xlabel("繰り返し回数: {\it n}");
ylabel("誤差: {\it |x_{n-1}-x_{n}|}")

legend('２分法','ニュートン法')


grid on;

File: make_graph.py
import pandas as pd
import matplotlib.pyplot as plt

# Load the data
bi_result = pd.read_csv("bi_result.csv", header=None)
new_result = pd.read_csv("new_result.csv", header=None)

# Configure the plotting environment
plt.rcParams.update({
    'font.size': 25,
    'font.family': 'Times New Roman',
    'axes.labelsize': 25,
    'axes.titlesize': 25
})

# Create the figure
fig, axs = plt.subplots(2, 1, figsize=(10, 12))

# First subplot
axs[0].plot(bi_result.iloc[:, 0], bi_result.iloc[:, 2], '-b', linewidth=2, label='Bisection Method')
axs[0].plot(new_result.iloc[:, 0], new_result.iloc[:, 2], '-r', linewidth=2, label='Newton Method')
axs[0].set_xlabel("Iteration Number: $n$")
axs[0].set_ylabel("Estimate: $x_n$")
axs[0].legend()
axs[0].grid(True)

# Second subplot
axs[1].semilogy(bi_result.iloc[:, 0], bi_result.iloc[:, 1], '-b', linewidth=2, label='Bisection Method')
axs[1].semilogy(new_result.iloc[:, 0], new_result.iloc[:, 1], '-r', linewidth=2, label='Newton Method')
axs[1].set_xlabel("Iteration Number: $n$")
axs[1].set_ylabel("Error: $|x_{n-1} - x_n|$")
axs[1].legend()
axs[1].grid(True)

# Display the plot
plt.tight_layout()
plt.show()
